### PR TITLE
[Feature] Support sided bar

### PR DESCRIPTION
**Pull request by hdygxsj**
### Purpose
Add sided bar for the front-end
image

### CODE REVIEW

**Comment by s7monk** [comment]
image
@hdygxsj Thank you for your contribution，I think the header navigation can be done without icons, which will make it look more concise.

File: paimon-web-ui-new/src/api/models/login/index.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ specific language governing permissions and limitations
 under the License. */
 
 import httpRequest from '../../request'
-import type ResponseOptions from '@/api/types'
+import type {ResponseOptions} from '@/api/types'
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/api/models/login/index.ts:19]: code indentation
```suggestion
import type { ResponseOptions } from '@/api/types'
```

File: paimon-web-ui-new/src/layouts/content/components/menubar/index.tsx [hunk 2]
```diff
@@ -20,40 +20,15 @@ import { RouterLink } from "vue-router"
 import type { NavBar } from "@/store/config/type"
 
 export default defineComponent({
+  props:{
+    menuOptions: {
+      type: Array as PropType<any>,
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/layouts/content/components/menubar/index.tsx:25]: please use the naive Menu component Type
```suggestion
      type: Array as PropType<MenuOption>,
```

File: paimon-web-ui-new/src/layouts/content/components/sidebar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+import { defineComponent, ref, type PropType } from 'vue'
+import {  useRouter } from 'vue-router'
+import type { Router } from 'vue-router'
+
+const Sidebar = defineComponent({
+  name: 'Sidebar',
+  props: {
+    sideMenuOptions: {
+      type: Array as PropType<any>,
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/layouts/content/components/sidebar/index.tsx:26]:
```suggestion
      type: Array as PropType<MenuOption>,
```

File: paimon-web-ui-new/src/layouts/content/components/topbar/index.tsx [hunk 4]
```diff
@@ -21,6 +21,12 @@ import MenuBar from '../menubar'
 import ToolBar from '../toolbar'
 
 export default defineComponent({
+  props:{
+    headerMenuOptions: {
+      type: Array as PropType<any>,
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/layouts/content/components/topbar/index.tsx:26]:
```suggestion
      type: Array as PropType<MenuOption>,
```

File: paimon-web-ui-new/src/layouts/content/index.tsx [hunk 5]
```diff
@@ -17,21 +17,47 @@ under the License. */
...
 
 export default defineComponent({
   name: 'ContentPage',
   setup() {
+    const configStore = useConfigStore()
+    const route = useRoute()
+    const { menuOptions, state } = useData()
+    const getSideOption = (state: any) => {
+      const activeNavKey = configStore.getCurrentNavActive
+      state.sideMenuOptions = menuOptions.value.filter((m: any) => m.key === activeNavKey)[0]?.sideMenuOptions || []
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/layouts/content/index.tsx:32]: replace `filter` to `find`

File: paimon-web-ui-new/src/layouts/content/components/topbar/index.tsx (hunk 4 above)
Comment (hdygxsj) [reply paimon-web-ui-new/src/layouts/content/components/topbar/index.tsx]: done

File: paimon-web-ui-new/src/layouts/content/index.tsx [hunk 6]
```diff
@@ -17,21 +17,47 @@ under the License. */
...
 export default defineComponent({
   name: 'ContentPage',
   setup() {
+    const configStore = useConfigStore()
+    const route = useRoute()
+    const { menuOptions, state } = useData()
+    const getSideOption = (state: any) => {
+      const activeNavKey = configStore.getCurrentNavActive
+      state.sideMenuOptions = menuOptions.value.find((m: any) => m.key === activeNavKey)?.sideMenuOptions || []
+      state.isShowSided = state.sideMenuOptions && state.sideMenuOptions.length > 0
```
Comment (lq-c) [inline paimon-web-ui-new/src/layouts/content/index.tsx:33]:
```suggestion
      state.isShowSided = state.sideMenuOptions && state.sideMenuOptions.length
```

File: paimon-web-ui-new/vite.config.ts [hunk 7]
```diff
@@ -29,7 +29,7 @@ export default defineConfig({
   server: {
     proxy: {
       '/api': {
-        target: 'http://47.94.247.86:10088',
+        target: 'http://127.0.0.1:10088',
```
Comment (lq-c) [inline paimon-web-ui-new/vite.config.ts:32]: This configuration does not need to be submitted

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**